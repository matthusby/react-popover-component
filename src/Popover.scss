$react-popover-component-font-color: #000 !default;
$react-popover-component-border-color: #9b9b9b !default;
$react-popover-component-border-radius: 4px !default;

@mixin add-arrow($position: top, $side: left) {
  $opposite: bottom !default;
  $horizontal: $side !default;

  @if $position == bottom { $opposite: top; }
  @if $position == left { $opposite: right; }
  @if $position == right { $opposite: left; }

  // Stroke
  &:before {
    #{$position}: -11px;
    border-#{$opposite}-color: $react-popover-component-border-color;
    border-#{$position}: 0;
    border-width: 11px;
    @if $side != center {
      #{$horizontal}: 21px;
    }
  }

  &:before, &:after {
    @if $side == center {
      @if $position == bottom or $position == top {
        transform: translateX(-50%);
        left: 50%;
      } @else {
        transform: translateY(-50%);
        top: 50%;
      }
    }
  }

  // Fill
  &:after {
    #{$position}: -8px;
    border-#{$opposite}-color: #fff;
    border-#{$position}: 0;
    border-width: 10px;
    @if $side != center {
      #{$horizontal}: 22px;
    }
  }
}

.react-popover-container {
  position: relative;
  width: 100%;
}

.react-popover {
  background: #fff;
  color: $react-popover-component-font-color;
  display: block;
  font-family: inherit;
  font-weight: normal;
  position: absolute;
  top: 0;
  z-index: 8;
  border: 2px solid $react-popover-component-border-color;
  border-radius: $react-popover-component-border-radius;

  .react-popover-wrapper {
    padding: 10px;
    display: flex;
    flex-direction: column;
    padding: rem-calc(18 20 24);
    padding-bottom: 1.25rem;
    text-align: left;

    p {
      margin-bottom: 0;
    }
    .button {
      margin-bottom: 0 !important;
    }
  }

  .react-popover-close {
    color: $react-popover-component-font-color !important;
    font-size: 14px;
    font-weight: normal;
    line-height: .5 !important;
    position: absolute;
    text-decoration: none;
    top: 3px;
    right: 3px;

    &:hover,
    &:focus {
      color: $react-popover-component-font-color !important;
    }
  }

  .react-popover-cancel {
    margin-top: 5px;
  }

  .react-popover-cta {
    align-items: center;
    display: flex;
    justify-content: flex-end;
    margin-top: 1rem;
  }

  .heading {
    color: $react-popover-component-font-color;
    margin-bottom: .5rem;
  }

  // Popover with arrow
  &.arrow {
    // generic style for arrow
    &:before,
    &:after {
      border-color: transparent;
      border-style: solid;
      content: '';
      height: 0;
      position: absolute;
      width: 0;
    }

    &.arrow-bottom-right {
      @include add-arrow(bottom, right);
    }

    &.arrow-bottom-left {
      @include add-arrow(bottom, left);
    }

    &.arrow-top-right {
      @include add-arrow(top, right);
    }

    &.arrow-top-left {
      @include add-arrow(top, left);
    }

    &.arrow-left-top {
      @include add-arrow(left, top);
    }

    &.arrow-left-bottom {
      @include add-arrow(left, bottom);
    }

    &.arrow-right-top {
      @include add-arrow(right, top);
    }

    &.arrow-right-bottom {
      @include add-arrow(right, bottom);
    }

    &.arrow-top-center {
      @include add-arrow(top, center);
    }

    &.arrow-bottom-center {
      @include add-arrow(bottom, center);
    }

    // TODO:
    &.arrow-left-center {
      @include add-arrow(left, center);
    }

    // TODO:
    &.arrow-right-center {
      @include add-arrow(right, center);
    }
  }
}

.react-popover-body {
  // make the clickable target area no larger than the child
  display: inline-block;
  position: relative;
  width: 100%;

  .react-popover-target-wrapper {
    display: inline-block;
  }
}

// Styles for screens that are at least 768px;
@media screen and (min-height: 768px) {
  .react-popover {
    width: 300px;
    float: inherit;
  }
}
