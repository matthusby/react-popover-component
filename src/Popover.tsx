import * as React from 'react';
import { Component } from 'react';
import onClickOutside from 'react-onclickoutside';
import config, { Arrows } from './config';
import TransformCalculator from './transform-calculator';

import { IProps, IState, IViewport } from './types';

class Popover extends Component<IProps, IState> {
  public static defaultProps: Partial<IProps> = {
    children: (<div>{config.warnings.missingContent}</div>),
    arrowPosition: 'bottomCenter',
    hasCloseButton: false,
    hasCancelLink: false,
    style: {},
    open: false,
    toggable: true,
    closeOnOutsideClick: true,
    onClose: undefined
  };

  private container: HTMLDivElement;
  private _transform: TransformCalculator;

  constructor(props) {
    super(props);

    this.state = {
      open: props.open
    };
  }

  handleClickOutside = (evt): void => {
    if (this.props.closeOnOutsideClick) this.close();
  }

  componentWillReceiveProps(nextProps: IProps): void {
    if('open' in nextProps) {
      this.setState({ open: nextProps.open });
    }
  }

  componentDidMount(): void {
    window.addEventListener('keyup', this.onEscapeKey, false);
    window.addEventListener('resize', this.handleResize);
    this.setState({ viewport: this.viewport() });
  }

  componentWillUnmount(): void {
    window.removeEventListener('keyup', this.onEscapeKey, false);
    window.removeEventListener('resize', this.handleResize);
  }

  viewport = (): IViewport => {
    return {
      width: window.innerWidth,
      height: window.innerHeight
    };
  }

  handleResize = (): void => {
    const viewport = this.viewport();
    this.setState({ viewport });
  }

  onEscapeKey = (e): void => {
    if (e.key === 'Escape') this.close();
  }

  onTargetClick = (): void => {
    if (!this.props.toggable) return;
    this.state.open ? this.close() : this.open();
  }

  onCancelClick = (e): void => {
    e.preventDefault();
    this.close();
  }

  setDefaultArrowClass(): string {
    const { arrowPosition } = this.props;
    return config.popOverStyles[arrowPosition] || config.popOverStyles.default;
  }

  open = (): void => this.setState({ open: true });

  close = (): void => {
    this.setState({ open: false });

    if (this.props.onClose) { this.props.onClose() }
  }

  containerWidth = (): string => {
    return this.container.clientWidth + 'px';
  }

  containerOffsets = (): ClientRect => {
    return this.container.getBoundingClientRect();
  }

  widthInsideViewport = (): string => {
    const { width } = this.props.style;
    const { viewport } = this.state;

    if (!width || !viewport) return this.containerWidth();

    const elementWidth = parseInt(width.replace('px', ''));
    const offsets = this.containerOffsets();

    switch (this.props.arrowPosition) {
      case 'topCenter':
      case 'bottomCenter':
        const elementLeft = this.transform().containerTranslateX();
        const actualWidth = offsets.left + elementLeft + elementWidth / 2;
        if (actualWidth > viewport.width) return this.containerWidth();
        break;
    }

    return width;
  }

  transform = (): TransformCalculator => {
    if (!this._transform) this._transform = new TransformCalculator(this.props.arrowPosition, this.container);

    return this._transform;
  }

  transformStyle = (): string => {
    const { left, top } = this.props;

    const x = left ? `translateX(${left})` : this.transform().translateX();
    const y = top ? `translateY(${top})` : this.transform().translateY();

    return `${x} ${y}`;
  }

  popoverStyles() {
    const { style } = this.props;

    if (!this.container) return style;

    style.width = this.widthInsideViewport();

    const transform = this.transformStyle();

    return { ...style, transform };
  }

  renderCloseButton(): JSX.Element|null {
    if (!this.props.hasCloseButton) return null;

    return (
      <a className="react-popover-close" href="" onClick={this.onCancelClick}>
        <span className="fa fa-close" />
      </a>
    );
  }

  renderCancelLink(): JSX.Element|null {
    if (!this.props.hasCancelLink) return null;

    return (
      <a className="react-popover-cancel" href="" onClick={this.onCancelClick} >
        Cancel
      </a>
    );
  }

  renderPopoverBody(): JSX.Element|null {
    const { children, actionButton } = this.props;

    const arrowClass = this.setDefaultArrowClass();
    if (!this.state.open) return null;

    return (
      <div
        className={`react-popover arrow ${arrowClass}`}
        style={{...this.popoverStyles()}}
      >
        <div className="react-popover-wrapper">
          {children}
          <div className={!!actionButton ? "react-popover-cta" : "" }>
            {actionButton}
            {this.renderCancelLink()}
          </div>
          {this.renderCloseButton()}
        </div>
      </div>
    );
  }

  setupContainer = (ref): void => {
    if (ref) this.container = ref;
  }

  render(): JSX.Element {
    const { className, target } = this.props;

    return (
      <div
        className={`react-popover-container ${className}`}
        ref={this.setupContainer}
      >
        <div className="react-popover-body">
          <div className="react-popover-target-wrapper" onClick={this.onTargetClick}>
            {target}
          </div>
          {this.renderPopoverBody()}
        </div>
      </div>
    );
  }
};

export default onClickOutside(Popover);
