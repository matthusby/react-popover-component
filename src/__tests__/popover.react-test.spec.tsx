import * as React from "react";
import { mount } from 'enzyme';
import Popover from '../Popover';

let props = {
  children: React.createElement('p', null, 'popover message'),
  top: '100px',
  left: '200px',
  arrowPosition: 'bottomRight',
  target: React.createElement('button',
    { className: 'btn-delete', onClick: (e) => { e.preventDefault(); } },
    'delete'),
  actionButton: React.createElement('button',
    { className: 'btn-confirm-delete', onClick: () => {} },
    'delete'),
  hasCloseButton: true,
  hasCancelLink: true,
};

const renderComponent = (props) => {
  return mount(<Popover {...props} />);
}

describe('Popover', () => {
  it('renders the popover with hide class applied by default', () => {
    const tree = renderComponent(props);
    expect(tree).toMatchSnapshot();
  });

  it('renders the popover without the hide class when open props is passed in', () => {
    props = { ...props, open: true };
    const tree = renderComponent(props);
    expect(tree).toMatchSnapshot();
  });

  it('renders with different arrowPosition', () => {
    props = { ...props, arrowPosition: 'leftTop' };
    const tree = renderComponent(props);
    expect(tree).toMatchSnapshot();
  });

  describe('popover body transform propery', () => {
    describe('when arrowPosition is topCenter', () => {
      beforeEach(() => {
        props = { ...props, arrowPosition: 'topCenter' };
      });

      it('renders popover with correct transform style', () => {
        const tree = renderComponent(props);
        expect(tree).toMatchSnapshot();
      });
    });

    describe('when arrowPosition is topRight', () => {
      beforeEach(() => {
        props = { ...props, arrowPosition: 'topRight' };
      });

      it('renders popover with correct transform style', () => {
        const tree = renderComponent(props);
        expect(tree).toMatchSnapshot();
      });
    });

    describe('otherwise', () => {
      beforeEach(() => {
        props = { ...props, arrowPosition: 'rightBottom' };
      });

      it('renders popover with correct transform style', () => {
        const tree = renderComponent(props);
        expect(tree).toMatchSnapshot();
      });
    });
  });
});
