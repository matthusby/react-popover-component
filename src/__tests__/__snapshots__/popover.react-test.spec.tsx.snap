// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Popover popover body transform propery otherwise renders popover with correct transform style 1`] = `
<OnClickOutside(Popover)
  actionButton={
    <button
      className="btn-confirm-delete"
      onClick={[Function]}
    >
      delete
    </button>
  }
  arrowPosition="rightBottom"
  eventTypes={
    Array [
      "mousedown",
      "touchstart",
    ]
  }
  excludeScrollbar={false}
  hasCancelLink={true}
  hasCloseButton={true}
  left="200px"
  open={true}
  outsideClickIgnoreClass="ignore-react-onclickoutside"
  preventDefault={false}
  stopPropagation={false}
  target={
    <button
      className="btn-delete"
      onClick={[Function]}
    >
      delete
    </button>
  }
  top="100px"
>
  <Popover
    actionButton={
      <button
        className="btn-confirm-delete"
        onClick={[Function]}
      >
        delete
      </button>
    }
    arrowPosition="rightBottom"
    closeOnOutsideClick={true}
    disableOnClickOutside={[Function]}
    enableOnClickOutside={[Function]}
    eventTypes={
      Array [
        "mousedown",
        "touchstart",
      ]
    }
    hasCancelLink={true}
    hasCloseButton={true}
    left="200px"
    open={true}
    outsideClickIgnoreClass="ignore-react-onclickoutside"
    preventDefault={false}
    stopPropagation={false}
    style={
      Object {
        "width": "0px",
      }
    }
    target={
      <button
        className="btn-delete"
        onClick={[Function]}
      >
        delete
      </button>
    }
    toggable={true}
    top="100px"
  >
    <div
      className="react-popover-container undefined"
    >
      <div
        className="react-popover-body"
      >
        <div
          className="react-popover-target-wrapper"
          onClick={[Function]}
        >
          <button
            className="btn-delete"
            onClick={[Function]}
          >
            delete
          </button>
        </div>
        <div
          className="react-popover arrow arrow-right-bottom"
          style={
            Object {
              "transform": "translateX(200px) translateY(100px)",
              "width": "0px",
            }
          }
        >
          <div
            className="react-popover-wrapper"
          >
            <p>
              popover message
            </p>
            <div
              className="react-popover-cta"
            >
              <button
                className="btn-confirm-delete"
                onClick={[Function]}
              >
                delete
              </button>
              <a
                className="react-popover-cancel"
                href=""
                onClick={[Function]}
              >
                Cancel
              </a>
            </div>
            <a
              className="react-popover-close"
              href=""
              onClick={[Function]}
            >
              <span
                className="fa fa-close"
              />
            </a>
          </div>
        </div>
      </div>
    </div>
  </Popover>
</OnClickOutside(Popover)>
`;

exports[`Popover popover body transform propery when arrowPosition is topCenter renders popover with correct transform style 1`] = `
<OnClickOutside(Popover)
  actionButton={
    <button
      className="btn-confirm-delete"
      onClick={[Function]}
    >
      delete
    </button>
  }
  arrowPosition="topCenter"
  eventTypes={
    Array [
      "mousedown",
      "touchstart",
    ]
  }
  excludeScrollbar={false}
  hasCancelLink={true}
  hasCloseButton={true}
  left="200px"
  open={true}
  outsideClickIgnoreClass="ignore-react-onclickoutside"
  preventDefault={false}
  stopPropagation={false}
  target={
    <button
      className="btn-delete"
      onClick={[Function]}
    >
      delete
    </button>
  }
  top="100px"
>
  <Popover
    actionButton={
      <button
        className="btn-confirm-delete"
        onClick={[Function]}
      >
        delete
      </button>
    }
    arrowPosition="topCenter"
    closeOnOutsideClick={true}
    disableOnClickOutside={[Function]}
    enableOnClickOutside={[Function]}
    eventTypes={
      Array [
        "mousedown",
        "touchstart",
      ]
    }
    hasCancelLink={true}
    hasCloseButton={true}
    left="200px"
    open={true}
    outsideClickIgnoreClass="ignore-react-onclickoutside"
    preventDefault={false}
    stopPropagation={false}
    style={
      Object {
        "width": "0px",
      }
    }
    target={
      <button
        className="btn-delete"
        onClick={[Function]}
      >
        delete
      </button>
    }
    toggable={true}
    top="100px"
  >
    <div
      className="react-popover-container undefined"
    >
      <div
        className="react-popover-body"
      >
        <div
          className="react-popover-target-wrapper"
          onClick={[Function]}
        >
          <button
            className="btn-delete"
            onClick={[Function]}
          >
            delete
          </button>
        </div>
        <div
          className="react-popover arrow arrow-top-center"
          style={
            Object {
              "transform": "translateX(200px) translateY(100px)",
              "width": "0px",
            }
          }
        >
          <div
            className="react-popover-wrapper"
          >
            <p>
              popover message
            </p>
            <div
              className="react-popover-cta"
            >
              <button
                className="btn-confirm-delete"
                onClick={[Function]}
              >
                delete
              </button>
              <a
                className="react-popover-cancel"
                href=""
                onClick={[Function]}
              >
                Cancel
              </a>
            </div>
            <a
              className="react-popover-close"
              href=""
              onClick={[Function]}
            >
              <span
                className="fa fa-close"
              />
            </a>
          </div>
        </div>
      </div>
    </div>
  </Popover>
</OnClickOutside(Popover)>
`;

exports[`Popover popover body transform propery when arrowPosition is topRight renders popover with correct transform style 1`] = `
<OnClickOutside(Popover)
  actionButton={
    <button
      className="btn-confirm-delete"
      onClick={[Function]}
    >
      delete
    </button>
  }
  arrowPosition="topRight"
  eventTypes={
    Array [
      "mousedown",
      "touchstart",
    ]
  }
  excludeScrollbar={false}
  hasCancelLink={true}
  hasCloseButton={true}
  left="200px"
  open={true}
  outsideClickIgnoreClass="ignore-react-onclickoutside"
  preventDefault={false}
  stopPropagation={false}
  target={
    <button
      className="btn-delete"
      onClick={[Function]}
    >
      delete
    </button>
  }
  top="100px"
>
  <Popover
    actionButton={
      <button
        className="btn-confirm-delete"
        onClick={[Function]}
      >
        delete
      </button>
    }
    arrowPosition="topRight"
    closeOnOutsideClick={true}
    disableOnClickOutside={[Function]}
    enableOnClickOutside={[Function]}
    eventTypes={
      Array [
        "mousedown",
        "touchstart",
      ]
    }
    hasCancelLink={true}
    hasCloseButton={true}
    left="200px"
    open={true}
    outsideClickIgnoreClass="ignore-react-onclickoutside"
    preventDefault={false}
    stopPropagation={false}
    style={
      Object {
        "width": "0px",
      }
    }
    target={
      <button
        className="btn-delete"
        onClick={[Function]}
      >
        delete
      </button>
    }
    toggable={true}
    top="100px"
  >
    <div
      className="react-popover-container undefined"
    >
      <div
        className="react-popover-body"
      >
        <div
          className="react-popover-target-wrapper"
          onClick={[Function]}
        >
          <button
            className="btn-delete"
            onClick={[Function]}
          >
            delete
          </button>
        </div>
        <div
          className="react-popover arrow arrow-top-right"
          style={
            Object {
              "transform": "translateX(200px) translateY(100px)",
              "width": "0px",
            }
          }
        >
          <div
            className="react-popover-wrapper"
          >
            <p>
              popover message
            </p>
            <div
              className="react-popover-cta"
            >
              <button
                className="btn-confirm-delete"
                onClick={[Function]}
              >
                delete
              </button>
              <a
                className="react-popover-cancel"
                href=""
                onClick={[Function]}
              >
                Cancel
              </a>
            </div>
            <a
              className="react-popover-close"
              href=""
              onClick={[Function]}
            >
              <span
                className="fa fa-close"
              />
            </a>
          </div>
        </div>
      </div>
    </div>
  </Popover>
</OnClickOutside(Popover)>
`;

exports[`Popover renders the popover with hide class applied by default 1`] = `
<OnClickOutside(Popover)
  actionButton={
    <button
      className="btn-confirm-delete"
      onClick={[Function]}
    >
      delete
    </button>
  }
  arrowPosition="bottomRight"
  eventTypes={
    Array [
      "mousedown",
      "touchstart",
    ]
  }
  excludeScrollbar={false}
  hasCancelLink={true}
  hasCloseButton={true}
  left="200px"
  outsideClickIgnoreClass="ignore-react-onclickoutside"
  preventDefault={false}
  stopPropagation={false}
  target={
    <button
      className="btn-delete"
      onClick={[Function]}
    >
      delete
    </button>
  }
  top="100px"
>
  <Popover
    actionButton={
      <button
        className="btn-confirm-delete"
        onClick={[Function]}
      >
        delete
      </button>
    }
    arrowPosition="bottomRight"
    closeOnOutsideClick={true}
    disableOnClickOutside={[Function]}
    enableOnClickOutside={[Function]}
    eventTypes={
      Array [
        "mousedown",
        "touchstart",
      ]
    }
    hasCancelLink={true}
    hasCloseButton={true}
    left="200px"
    open={false}
    outsideClickIgnoreClass="ignore-react-onclickoutside"
    preventDefault={false}
    stopPropagation={false}
    style={Object {}}
    target={
      <button
        className="btn-delete"
        onClick={[Function]}
      >
        delete
      </button>
    }
    toggable={true}
    top="100px"
  >
    <div
      className="react-popover-container undefined"
    >
      <div
        className="react-popover-body"
      >
        <div
          className="react-popover-target-wrapper"
          onClick={[Function]}
        >
          <button
            className="btn-delete"
            onClick={[Function]}
          >
            delete
          </button>
        </div>
      </div>
    </div>
  </Popover>
</OnClickOutside(Popover)>
`;

exports[`Popover renders the popover without the hide class when open props is passed in 1`] = `
<OnClickOutside(Popover)
  actionButton={
    <button
      className="btn-confirm-delete"
      onClick={[Function]}
    >
      delete
    </button>
  }
  arrowPosition="bottomRight"
  eventTypes={
    Array [
      "mousedown",
      "touchstart",
    ]
  }
  excludeScrollbar={false}
  hasCancelLink={true}
  hasCloseButton={true}
  left="200px"
  open={true}
  outsideClickIgnoreClass="ignore-react-onclickoutside"
  preventDefault={false}
  stopPropagation={false}
  target={
    <button
      className="btn-delete"
      onClick={[Function]}
    >
      delete
    </button>
  }
  top="100px"
>
  <Popover
    actionButton={
      <button
        className="btn-confirm-delete"
        onClick={[Function]}
      >
        delete
      </button>
    }
    arrowPosition="bottomRight"
    closeOnOutsideClick={true}
    disableOnClickOutside={[Function]}
    enableOnClickOutside={[Function]}
    eventTypes={
      Array [
        "mousedown",
        "touchstart",
      ]
    }
    hasCancelLink={true}
    hasCloseButton={true}
    left="200px"
    open={true}
    outsideClickIgnoreClass="ignore-react-onclickoutside"
    preventDefault={false}
    stopPropagation={false}
    style={
      Object {
        "width": "0px",
      }
    }
    target={
      <button
        className="btn-delete"
        onClick={[Function]}
      >
        delete
      </button>
    }
    toggable={true}
    top="100px"
  >
    <div
      className="react-popover-container undefined"
    >
      <div
        className="react-popover-body"
      >
        <div
          className="react-popover-target-wrapper"
          onClick={[Function]}
        >
          <button
            className="btn-delete"
            onClick={[Function]}
          >
            delete
          </button>
        </div>
        <div
          className="react-popover arrow arrow-bottom-right"
          style={
            Object {
              "transform": "translateX(200px) translateY(100px)",
              "width": "0px",
            }
          }
        >
          <div
            className="react-popover-wrapper"
          >
            <p>
              popover message
            </p>
            <div
              className="react-popover-cta"
            >
              <button
                className="btn-confirm-delete"
                onClick={[Function]}
              >
                delete
              </button>
              <a
                className="react-popover-cancel"
                href=""
                onClick={[Function]}
              >
                Cancel
              </a>
            </div>
            <a
              className="react-popover-close"
              href=""
              onClick={[Function]}
            >
              <span
                className="fa fa-close"
              />
            </a>
          </div>
        </div>
      </div>
    </div>
  </Popover>
</OnClickOutside(Popover)>
`;

exports[`Popover renders with different arrowPosition 1`] = `
<OnClickOutside(Popover)
  actionButton={
    <button
      className="btn-confirm-delete"
      onClick={[Function]}
    >
      delete
    </button>
  }
  arrowPosition="leftTop"
  eventTypes={
    Array [
      "mousedown",
      "touchstart",
    ]
  }
  excludeScrollbar={false}
  hasCancelLink={true}
  hasCloseButton={true}
  left="200px"
  open={true}
  outsideClickIgnoreClass="ignore-react-onclickoutside"
  preventDefault={false}
  stopPropagation={false}
  target={
    <button
      className="btn-delete"
      onClick={[Function]}
    >
      delete
    </button>
  }
  top="100px"
>
  <Popover
    actionButton={
      <button
        className="btn-confirm-delete"
        onClick={[Function]}
      >
        delete
      </button>
    }
    arrowPosition="leftTop"
    closeOnOutsideClick={true}
    disableOnClickOutside={[Function]}
    enableOnClickOutside={[Function]}
    eventTypes={
      Array [
        "mousedown",
        "touchstart",
      ]
    }
    hasCancelLink={true}
    hasCloseButton={true}
    left="200px"
    open={true}
    outsideClickIgnoreClass="ignore-react-onclickoutside"
    preventDefault={false}
    stopPropagation={false}
    style={
      Object {
        "width": "0px",
      }
    }
    target={
      <button
        className="btn-delete"
        onClick={[Function]}
      >
        delete
      </button>
    }
    toggable={true}
    top="100px"
  >
    <div
      className="react-popover-container undefined"
    >
      <div
        className="react-popover-body"
      >
        <div
          className="react-popover-target-wrapper"
          onClick={[Function]}
        >
          <button
            className="btn-delete"
            onClick={[Function]}
          >
            delete
          </button>
        </div>
        <div
          className="react-popover arrow arrow-left-top"
          style={
            Object {
              "transform": "translateX(200px) translateY(100px)",
              "width": "0px",
            }
          }
        >
          <div
            className="react-popover-wrapper"
          >
            <p>
              popover message
            </p>
            <div
              className="react-popover-cta"
            >
              <button
                className="btn-confirm-delete"
                onClick={[Function]}
              >
                delete
              </button>
              <a
                className="react-popover-cancel"
                href=""
                onClick={[Function]}
              >
                Cancel
              </a>
            </div>
            <a
              className="react-popover-close"
              href=""
              onClick={[Function]}
            >
              <span
                className="fa fa-close"
              />
            </a>
          </div>
        </div>
      </div>
    </div>
  </Popover>
</OnClickOutside(Popover)>
`;
